### PR TITLE
Add context logic

### PR DESCRIPTION
**Pull request by Fralleee**
- Added context logic
  - Pizzas can be added to orders 
  - Context is saved in localStorage in case of page refresh
- Added some personal tweaks to eslint configuration 
- Applied styling rules to formatter for unified code style
- Added tests for business logic

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module.exports = {
+  extends: ["react-app", "prettier"],
```
Comment (vongohren) [inline .eslintrc.js:2]: Self made rulesett :D ?

File: src/businessLogic/bonusLogic.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  total: amount * (1 - (bonus?.deductionPercentage ?? 0)),
+  bonusOnly: amount * (bonus?.deductionPercentage ?? 0)
+})
+export const calculateBonus = (order) => {
+  const { gold, silver, bronze } = bonusLevels
+  const amount = order.reduce(
+    (acc, curr) => (acc += curr.cost * curr.amount),
+    0
+  )
+  return amount >= gold.threshold
```
Comment (vongohren) [inline src/businessLogic/bonusLogic.js:20]: 💯

File: src/businessLogic/orderLogic.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const pizzaOrderMapper = (pizza, amount = 1) => ({ name: pizza.name, size: pizza.size, cost: pizza.cost, amount })
+const identifier = pizza => pizza.name + pizza.size
+const sortByName = (a, b) => identifier(a) > identifier(b) ? 1 : (identifier(b) > identifier(a) ? -1 : 0)
+
+export const addPizza = (order, pizza) => {
```
Comment (vongohren) [inline src/businessLogic/orderLogic.js:5]: What is your thoughts on that this way of doing code can quickly become difficult to read?
Especially if someone is not used to this coding style

File: src/components/ShoppingCart.jsx [hunk 4]
```diff
@@ -14,24 +14,24 @@ const CartButton = styled(Button)`
...
-  const [origin, setOrigin] = useState({});
+  const [open, setOpen] = useState(false)
+  const [origin, setOrigin] = useState({})
 
   const openModal = e => {
-    const rect = e.currentTarget.getBoundingClientRect();
-    setOrigin({ left: rect.left - rect.width * 0.5, top: rect.top - rect.height * 0.5 });
-    setOpen(true);
+    const rect = e.currentTarget.getBoundingClientRect()
+    setOrigin({ left: rect.left - rect.width * 0.5, top: rect.top - rect.height * 0.5 })
```
Comment (vongohren) [inline src/components/ShoppingCart.jsx:25]: Placement of the modal on the screen?

File: src/components/order/Order.jsx [hunk 5]
```diff
@@ -32,28 +32,17 @@ const CloseButton = styled(Button)`
     font-size: 1.5rem;
   }
 `
-// #endregion
+//#endregion
```
Comment (vongohren) [inline src/components/order/Order.jsx:35]: Where is this style comingfrom? What does this help with? Something inside vscode?

File: src/tests/Bonus.test.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import { getBonusAmounts, calculateBonus } from "businessLogic/bonusLogic"
+import { bonusLevels } from "constants/bonusConstants"
+
+test("should calculate next available bonus without any applied bonus", () => {
```
Comment (vongohren) [inline src/tests/Bonus.test.js:4]: Nice 💯

**Review by vongohren — COMMENTED** [review]
Just some questions :D

File: .eslintrc.js (hunk 1 above)
Comment (Fralleee) [reply .eslintrc.js:2]: It is a npm package that: 
_Turns off all rules that are unnecessary or might conflict with Prettier._

Prettier is an extension I use to format the code, however I actually removed it later on in the project and instead format using ESLint instead, so this extended config should be removed.

File: src/businessLogic/orderLogic.js (hunk 3 above)
Comment (Fralleee) [reply src/businessLogic/orderLogic.js:5]: Yeah, agreed.

This is ugly. Basically, is that I want to use the name + size as primary key for pizzas in an order.
However, I could resolve this into a property named "ID" in my pizzaOrderMapper-function instead.

File: src/components/ShoppingCart.jsx (hunk 4 above)
Comment (Fralleee) [reply src/components/ShoppingCart.jsx:25]: I get the bounding center coordinates of the clicked object, in this case the pizza button.

Then, when the modal gets instantiated, it starts from these coordinates and animates out into the center of the screen.
So we get a very subtle "pop out from the button"-effect :)

File: src/components/order/Order.jsx (hunk 5 above)
Comment (Fralleee) [reply src/components/order/Order.jsx:35]: I believe this originates from C# but is now also present in VSCode editor for multiple languages.
It lets you define a region that you can then fold, makes it easier to work with larger files.
https://code.visualstudio.com/docs/editor/codebasics#_folding

File: .eslintrc.js (hunk 1 above)
Comment (vongohren) [reply .eslintrc.js:2]: I ment the whole setup, but thanks for the reply :D 
But the setup turns off stuff that conflict with prettier?
Okok

Comment (vongohren) [reply .eslintrc.js:2]: Good that you are effficient with linting :D

File: src/businessLogic/orderLogic.js (hunk 3 above)
Comment (vongohren) [reply src/businessLogic/orderLogic.js:5]: This comment is enough, no need to change :D 
Thanks for sharing
Can resolve when seen

File: src/components/ShoppingCart.jsx (hunk 4 above)
Comment (vongohren) [reply src/components/ShoppingCart.jsx:25]: A nice! I liked the UI, simple and clean!

File: src/components/order/Order.jsx (hunk 5 above)
Comment (vongohren) [reply src/components/order/Order.jsx:35]: Ah cool, thanks for feedback!

File: .eslintrc.js (hunk 1 above)
Comment (Fralleee) [reply .eslintrc.js:2]: Ah right, the react-app contains multiple "best practice"-rules which I then extend/override with some that are personal preference e.g. no semicolon required and similar ones :)

Comment (vongohren) [reply .eslintrc.js:2]: Cool cool!

**Pull request merged**